designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ProcessesResources.scala

```diff
@@ -260,22 +260,14 @@ class ProcessesResources(
   }
 
   private def fetchProcessStatesForProcesses(processes: List[BaseProcessDetails[Unit]])(implicit user: LoggedUser): Future[Map[String, ProcessState]] = {
-    import cats.instances.future._
-    import cats.instances.list._
-    import cats.syntax.traverse._
     processes.map(process => deploymentService.getProcessState(process.idWithName).map(status => process.name -> status))
       .sequence[Future, (String, ProcessState)].map(_.toMap)
   }
 
-  //This is temporary function to enriching process state data
-  //TODO: Remove it when we will support cache for state
-  private def enrichDetailsWithProcessState[PS: ProcessShapeFetchStrategy](process: BaseProcessDetails[PS]): BaseProcessDetails[PS] =
-    process.copy(state = deploymentManager(process.processingType).map(m => m.processStateDefinitionManager.processState(
-      m.processStateDefinitionManager.mapActionToStatus(process.lastAction.map(_.action))
-    )))
-
-  private def deploymentManager(processingType: ProcessingType): Option[DeploymentManager] =
-    typeToConfig.forType(processingType).map(_.deploymentManager)
+  private def enrichDetailsWithProcessState[PS: ProcessShapeFetchStrategy](process: BaseProcessDetails[PS])
```

Comment:
Maybe it's time to move this method to the DeploymentService / some state service?

Reply:
I think that now it can be just inlined. I don't want to put the logic of "orchestration" of fetching process from db + enriching into DeploymentService. Let it be responsible only for executing actions and providing states. Maybe when this orchestration logic will be too big, we can introduce some new service